### PR TITLE
luajit: backport softfloat ppc support

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

Maintainer: @milani

### CODE REVIEW

**Comment by neheb** [comment]
@BKPepe merge or remove powerpc support? I know you guys use this on a powerpc platform.

File: lang/luajit/Makefile [hunk 1]
```diff
@@ -24,7 +24,7 @@ define Package/luajit
  CATEGORY:=Languages
  TITLE:=LuaJIT
  URL:=https://www.luajit.org
- DEPENDS:=@(i386||x86_64||arm||armeb||aarch64||(powerpc&&HAS_FPU)||mips||mipsel||mips64)
+ DEPENDS:=@(i386||x86_64||arm||armeb||aarch64||powerpc||mips||mipsel||mips64)
```
Comment (neheb) [inline lang/luajit/Makefile:27]: given that this is basically everything except ARC, probably should just change this to @!arc

**Comment by BKPepe** [comment]
We would like to use LuaJIT on PowerPC, so we can use Knot Resolver by default instead of unbound, but there were still issues that prevent us from using it how we want even though we tried as many forks as there were in the past (https://github.com/moonjit/moonjit/issues/101,  https://github.com/openresty/luajit2/issues/118) 

Sadly, this support will be removed and unsupported soon, but I don't think there is anything we can do about it. :cry: 

I can give it a last try to see what happens.

**Comment by neheb** [comment]
I ran into the same compilation issues in that issue and fixed them here. I have not done any runtime testing on powerpc.

**Comment by neheb** [comment]
guess I should also ping @Cynerd

**Comment by BKPepe** [comment]
> I have not done any runtime testing on powerpc.

I did it right now. It seems like the Knot Resolver works with that on Turris 1.1, p8540 (Freescale P2020), OpenWrt 21.02.3. @paper42 Can you check it on your end as well? I still find it somehow weird.

**Review by BKPepe — APPROVED** [review]

**Comment by BKPepe** [comment]
That was fast. You could wait a little bit until it is tested by my colleague.

**Pull request merged**